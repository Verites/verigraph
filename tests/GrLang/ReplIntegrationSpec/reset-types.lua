function print_types()
  print('Types: ')
  GrLang.print_types()
  print()
end

print_types()
GrLang.reset_types()
print_types()

GrLang.add_node_type('N')
GrLang.add_edge_type('E', 'N', 'N')
print_types()

function foo()
  local a = Graph[[]]
  print(pcall(GrLang.reset_types))
end

foo()
GrLang.reset_types()
print_types()

GrLang.add_node_type('N')
GrLang.add_edge_type('E', 'N', 'N')
print_types()

b = Graph[[]]
print(pcall(GrLang.reset_types))
b = nil
GrLang.reset_types()
print_types()