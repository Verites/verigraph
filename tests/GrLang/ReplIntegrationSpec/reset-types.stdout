Types: 
node type N
edge type E: N -> N

false	Cannot reset types, some GrLang values still refer to the type graph.
Types: 

Types: 
node type N
edge type E: N -> N

false	Cannot reset types, some GrLang values still refer to the type graph.
Types: 

